regions_www/m/server.R

```diff
@@ -19,804 +19,1071 @@
+      return()
+    # Check if the event$id is called from the "new_region" polygons
+    if (grepl("new_region", event$id)) {
+      # Split the id to identify the region name
+      new_region <- strsplit(event$id, " ")[[1]][2]
+      if (is.null(new_region))
+        return()
+      new_data_dir <- file.path(data_regional_root, input$purpose, region$geography, new_region)
 
-  # Updates the bounding box (bb) to the current map bb unless the map is frozen
-  # Returns a bb
-  flows_bb <- reactive({
-    if(!input$freeze || is.null(helper$bb)){
-      helper$bb <<- map_bb()
+      if (region$data_dir != new_data_dir &&
+          file.exists(new_data_dir)) {
+        region$current <- new_region
+        region$data_dir <- new_data_dir
+        region$repopulate_region <- F
+        if (input$freeze)
+          # If we change the map data then lines should not be frozen to the old map data
+          updateCheckboxInput(session, "freeze", value = F)
+      }
+      return()
     }
-    helper$bb
-  })
+    e_lat_lng <- paste0(event$lat, event$lng)
 
-  # Set freeze checkbox to false when lines are route_network, otherwise to true
-  # Also disable freeze checkbox for route_network
-  observe({
-    # Build a reactive expression for lines
-    input$line_type
-    # Also when user moves to a new region
-    region$repopulate_region
+    # Fix bug when a line has been clicked then the click event is re-emmited when the map is moved
+    if (e_lat_lng == helper$e_lat_lng)
+      return()
+    helper$e_lat_lng <<- e_lat_lng
 
-    if (! input$line_type %in% show_no_lines){
-      if (input$line_type == "rnet"){
-        updateCheckboxInput(session, "freeze", value = T)
-        disable("freeze")
-      }
-      else if (input$line_type != "rnet" && isolate(input$freeze)){
-        updateCheckboxInput(session, "freeze", value = F)
-        enable("freeze")
+    # Highlighting for the pop-ups
+    isolate({
+      id_line_type <- unlist(strsplit(event$id, "-"))
+      id <- id_line_type[1]
+      line_type <- id_line_type[2]
+
+      if (event$group == "centroids") {
+        addPolygons(
+          leafletProxy("map"),
+          data = to_plot$centroids[to_plot$c$geo_code == id, ],
+          fill = F,
+          color = get_line_colour("centroids") ,
+          opacity = 0.7,
+          layerId = "highlighted"
+        )
+      } else if (event$group == "zones") {
+        addPolygons(
+          leafletProxy("map"),
+          data = to_plot$zones[to_plot$z$geo_code == id, ],
+          fill = FALSE,
+          color = "black",
+          opacity = 0.7 ,
+          layerId = "highlighted"
+        )
+      } else {
+        line <- to_plot[[line_type]][to_plot[[line_type]]$id == id, ]
+        if (!is.null(line))
+          addPolylines(
+            leafletProxy("map"),
+            data = line,
+            color = "white",
+            opacity = 0.4,
+            layerId = "highlighted"
+          )
```

Comment:
Great, I'm assuming that most of the plotting code hasn't changed much (except checking if the data exists before trying to plot it).

Reply:
Fair assumption.

Reply:
Can we confirm this with @AnnaGoodman1 ?

Reply:
yes, I think that's right. The only real change I can think of is defining bins by purpose, since so many  areas have >40% ttravel to school in the school layer - hence adding in one expert level for that:

      # Define bins and breaks (by purpose)
      if (input$purpose == "commute" || input$purpose=="alltrips") {
        zbins <- zbins_commute
        zbreaks <- zbreaks_commute
      } else if (input$purpose == "school") {
        zbins <- zbins_school
        zbreaks <- zbreaks_school
      }